src/AllensCausalModel/CausalModels.jl

```diff
@@ -0,0 +1,57 @@
+module CausalModels
+using Distributions
+using Statistics
+export ExogenousVariable, EndogenousVariable, intervene, randomsample, prob
+	#represent as function with id
+	function ExogenousVariable(name,dist)
+		id=rand(1)
```

Comment:
- `rand(1)` returns a singleton vector of a 64 bit floating point number (a Float64).
  - Generally Floating point numbers are not a good idea to use as ids. Use symbols, or integers.  `rand(Int)` will give you a random integer.
 - There are several instances of `rand`.  I think it;s unlikely you wanted to return a vector.

- using a random number for an id is not a good idea.  Even though it is very unlikely, you might get a clash.  Use a counter instead.

- Instead of having id generated here, it would be better to have it as a keyword argument with a default argument:

`function ExogenousVariable(name, dist; id = newid())`